cumulusci/robotframework/Salesforce.py

```diff
@@ -499,6 +506,111 @@ def salesforce_insert(self, obj_name, **kwargs):
+                |  {'Name': 'User 2', 'Age': '2'}]
+
+            Python Expression Syntax is allowed so computed templates like this are also allowed: ``{{1000 + number}}``
+
+            Python operators can be used, but no functions or variables are provided, so mostly you just
+            have access to mathematical and logical operators. The Python operators are described here:
+
+            https://www.digitalocean.com/community/tutorials/how-to-do-math-in-python-3-with-operators
+
+            Contact the CCI team if you have a use-case that
+            could benefit from more expression language power.
+
+            Templates can also be based on faker patterns like those described here:
+
+            https://faker.readthedocs.io/en/master/providers.html
+
+            Most examples can be pasted into templates verbatim:
+
+                | @{objects}=  Generate Test Data  Contact  200
+                | ...  Name={{fake.first_name}} {{fake.last_name}}
+                | ...  MailingStreet={{fake.street_address}}
+                | ...  MailingCity=New York
+                | ...  MailingState=NY
+                | ...  MailingPostalCode=12345
+                | ...  Email={{fake.email(domain="salesforce.com")}}
+           """
+        objs = []
+
+        for i in range(int(number_to_create)):
+            formatted_fields = {
+                name: format_str(value, i) for name, value in fields.items()
+            }
+            newobj = self._salesforce_generate_object(obj_name, **formatted_fields)
+            objs.append(newobj)
+
+        return objs
+
+    def salesforce_collection_insert(self, objects):
+        """Inserts up to 200 records that were created with Generate Test Data.
+           The 200 record limit is enforced by the Salesforce APIs"""
+        assert (
+            not obj.get("id", None) for obj in objects
+        ), "Insertable objects should not have IDs"
+        assert len(objects) <= SF_COLLECTION_INSERTION_LIMIT, (
+            "Cannot insert more than %s objects with this keyword"
+            % SF_COLLECTION_INSERTION_LIMIT
+        )
+
+        records = self.cumulusci.sf.restful(
+            "composite/sobjects",
+            method="POST",
+            json={"allOrNone": True, "records": objects},
+        )
+
+        for record, obj in zip(records, objects):
+            if record["errors"]:
+                raise AssertionError(*record["errors"])
+            self.store_session_record(obj["attributes"]["type"], record["id"])
+            obj["id"] = record["id"]
+            obj[STATUS_KEY] = record
+
+        return objects
+
+    def salesforce_collection_update(self, objects):
+        """Updates up to 200 records described as Robot/Python dictionaries"""
+        for obj in objects:
+            assert obj[
+                "id"
+            ], "Should be a list of objects with Ids returned by Salesforce Collection Insert"
+            del obj[STATUS_KEY]
```

Comment:
I think this could be removed since it gets re-set below. Then this can also work with a list of objects that's been constructed in some other way and doesn't have statuses.

Reply:
I think I deleted it so that it won't get sent to Salesforce. I'd assume that either the client or the server side complains about it. I can make the deletion conditional though.

Reply:
Oh yeah, makes sense.

Reply:
https://github.com/SFDO-Tooling/CumulusCI/blob/feature/collection-apis/cumulusci/robotframework/Salesforce.py#L602